src/middleware/companyAuth.middleware.ts

```diff
@@ -0,0 +1,94 @@
+import ApplicationLogger from 'ch-logging/lib/ApplicationLogger'
+import { Session } from 'ch-node-session-handler'
+import { CookieConfig } from 'ch-node-session-handler/lib/config/CookieConfig'
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey'
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys'
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+
+import { Mutable } from 'app/models/mutable'
+import { generateNonce, jweEncodeWithNonce } from 'app/utils/jwt.encryption'
+
+export interface AuthConfig {
+  accountUrl: string,
+  accountRequestKey: string,
+  accountClientId: string,
+  chsUrl: string,
+}
+
+interface ISignInInfoWithCompanyNumber extends ISignInInfo {
```

Comment:
Seems like this should be added to the node session handler library as this is common functionality and not dissolutions specific